### ISSUE TITLE
add save_png for evaluation

### ISSUE THREAD

**Issue by Nic-Ma** [Project-MONAI/MONAI#318]
**Is your feature request related to a problem? Please describe.**
Now we already have `save nifiti` and `save classification`, also need to implement save png API.

**Comment by Bala93** [comment]
Nifti file save feature savea multiple channels(h,w,d) in a single file. How should the same be handled for png files. If more specifics are provided, I can take this up.

**Comment by Nic-Ma** [comment]
Hi @Bala93 ,

Thanks very much for your interest here.
The PNG saver only needs to support 2D data. Here are the basic task requirements:
1. Refer to the `nifiti_saver` and `write_nifti` APIs and implement `png_saver` and `write_png`.
2. The metadata only needs `filename_or_obj` and `spatial_shape` to  resize and save.
3. Add a `scale_factor` parameter, like:
```py
save_png(..., scale_factor=255):
    if scale_factor > 0:
        max_val = np.max(data)
        min_val = np.min(data)
        data = scale_factor * ((data - min_val) / (max_val - min_val))
```
4. (optional) move `_create_file_basename` function from nifti_saver to utils.py.

Thanks.

**Comment by wyli** [comment]
fixed by a preliminary version in #326 #357
